### PR TITLE
MINOR: Remove deprecated Resource class constructor usage

### PR DESCRIPTION
**Pull request by omkreddy**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@ijuma Pls take a look at this minor cleanup.

File: core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala [hunk 1]
```diff
@@ -263,7 +264,7 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
     AclCommand.main(produceAclArgs(tp.topic))
     AclCommand.main(consumeAclArgs(tp.topic))
     servers.foreach { s =>
-      TestUtils.waitAndVerifyAcls(TopicReadAcl ++ TopicWriteAcl ++ TopicDescribeAcl ++ TopicCreateAcl, s.apis.authorizer.get, new Resource(Topic, tp.topic))
+      TestUtils.waitAndVerifyAcls(TopicReadAcl ++ TopicWriteAcl ++ TopicDescribeAcl ++ TopicCreateAcl, s.apis.authorizer.get, new Resource(Topic, tp.topic, PatternType.LITERAL))
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala]: Please break the long line.

**Review by ijuma — COMMENTED** [review]
Just one trivial suggestion, looks good otherwise.

File: core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala (hunk 1 above)
Comment (omkreddy) [reply core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala]: @ijuma Thanks for the review. Updated the PR.

**Comment by omkreddy** [comment]
retest this please

**Review by ijuma — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
Test failures are unrelated.

**Comment by omkreddy** [comment]
test failures are not related.

@ijuma Can we merge this minor PR?

**Pull request merged**